# HG changeset patch
# Parent  44831e2cebd4a1f5422b9aec6ce48cc3700c6abd

diff -r 44831e2cebd4 svnkit-dav/src/main/java/org/tmatesoft/svn/core/internal/server/dav/DAVServlet.java
--- a/svnkit-dav/src/main/java/org/tmatesoft/svn/core/internal/server/dav/DAVServlet.java	Mon Feb 06 13:56:38 2017 +0100
+++ b/svnkit-dav/src/main/java/org/tmatesoft/svn/core/internal/server/dav/DAVServlet.java	Mon Feb 06 13:57:01 2017 +0100
@@ -131,7 +131,6 @@
             handler = DAVHandlerFactory.createHandler(repositoryManager, request, response);
             handler.execute();
         } catch (DAVException de) {
-            response.setContentType(XML_CONTENT_TYPE);
             handleError(de, response);
         } catch (SVNException svne) {
             StringWriter sw = new StringWriter();
@@ -253,6 +252,12 @@
                 servletResponse.getWriter().print(errorMessageBuffer.toString());
                 SVNDebugLog.getDefaultLog().logFine(SVNLogType.NETWORK, errorMessageBuffer.toString());
                 return;
+            } else {
+                // override application/xml content-type, for errors without xml body,
+                // because the svnclient will try to parse the response body, if the content type
+                // is application/xml. We will use text/html, because some application server will
+                // will return html pages for servlet error codes.
+                servletResponse.setContentType("text/html");
             }
             servletResponse.setStatus(error.getResponseCode());
             return;
